src/tool/hpcrun/control-knob.c

```diff
@@ -62,20 +66,18 @@ control_knob_init()
   char *in = getenv("HPCRUN_CONTROL_KNOBS");
   if (in == NULL) return;
 
-  char *save;
+  char *save = NULL;
   for (char *f = start_tok(in); more_tok(); f = next_tok()){
     char *tmp = strdup(f);
     char *name = strtok_r(tmp, "=", &save);
-    control_knob_t *iter = NULL;
+    char *value = strtok_r(NULL, "=", &save);
```

Comment:
This does not look right at all.

Reply:
@blue42u Can you explain why this isn't correct?

Reply:
Ignore that, I misremembered how `strtok` works. Looks fine.